### ISSUE TITLE
cannot install python libs to custom prefix

### ISSUE THREAD

**Issue by pescobar** [pandegroup/openmm#1428]
Hi 

I am compiling OpenMM from sources with a user without root access. I can compile fine and when I do `make && make install` the installation is done in my custom prefix `/tmp/openmm` but when I try to do `make PythonInstall` I get an error because the python module cannot be installed in the central python installation

```
running install
running build
running build_py
copying simtk/openmm/version.py -> build/lib.linux-x86_64-2.7/simtk/openmm
running build_ext
running install_lib
creating /scicore/soft/apps/Python/2.7.5-goolf-1.4.10/lib/python2.7/site-packages/simtk
error: could not create '/scicore/soft/apps/Python/2.7.5-goolf-1.4.10/lib/python2.7/site-packages/simtk': Permission denied
[100%] Built target PythonInstall
```

how can I install the python libraries to my custom installation prefix?

thanks

**Comment by peastman** [comment]
I gather you want to install it for use with the system Python?  (Are you sure you really want to be using Python 2.7?  It'll work, but that's really out of date.  If you install your own Python into your home directory, you can use something more modern, and avoid this problem at the same time.)

Assuming that's correct, follow the instructions at http://docs.openmm.org/6.3.0/userguide/library.html#installing-on-linux-and-mac.  That describes how to install by running the `setup.py` script directly.  When you get to the final step, add the `--user` option:

```
python setup.py install --user
```

That tells distutils to install the module into a directory inside your home directory.

**Comment by pescobar** [comment]
hi @peastman  thanks for your help

after following your suggestion I am getting this error. Do you know what could be the reason?

```
running build_ext
building 'simtk.openmm._openmm' extension
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/src
creating build/temp.linux-x86_64-2.7/src/swig_doxygen
gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -O2 -march=native -I/scicore/soft/apps/OpenBLAS/0.2.6-gompi-1.4.10-LAPACK-3.4.2/include -I/scicore/soft/apps/ScaLAPACK/2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2/include -I/scicore/soft/apps/FFTW/3.3.3-gompi-1.4.10/include -I/scicore/soft/apps/Python/2.7.5-goolf-1.4.10/include -fPIC -DMAJOR_VERSION=@OPENMM_MAJOR_VERSION@ -DMINOR_VERSION=@OPENMM_MINOR_VERSION@ -I/tmp/openmm/include/ -I/scicore/soft/apps/Python/2.7.5-goolf-1.4.10/include/python2.7 -c src/swig_doxygen/OpenMMSwig.cxx -o build/temp.linux-x86_64-2.7/src/swig_doxygen/OpenMMSwig.o
gcc: error: src/swig_doxygen/OpenMMSwig.cxx: No such file or directory
gcc: fatal error: no input files
compilation terminated.
```

**Comment by peastman** [comment]
At what point do you get that error?  Have you built everything else first?

**Comment by pescobar** [comment]
I could manage to install the python API. I thought that as I was doing `make PythonInstall` I didn't need to run `python setup.py build` again but I was wrong. I solved it by doing `python setup.py build && python setup.py install --prefix=/path/to/my/prefix`

I have contributed and easybuild recipe which reproduces my build. https://github.com/hpcugent/easybuild-easyconfigs/pull/2762

thanks for your help @peastman . I think this issue can be closed.
